opal_server/server.py

```diff
@@ -270,9 +270,15 @@ async def start_server_background_tasks(self):
                                     topic=opal_server_config.POLICY_REPO_WEBHOOK_TOPIC)
                         # init policy watcher
                         if self.watcher is None:
-                            # only the leader should discard previous clones
-                            self.create_local_clone_path_and_discard_previous_clones()
-                            self.watcher = setup_watcher_task(self.publisher, remote_source_url=self._policy_remote_url)
+                            clone_path_finder = RepoClonePathFinder(
+                                base_clone_path=opal_server_config.POLICY_REPO_CLONE_PATH,
+                                clone_subdirectory_prefix=opal_server_config.POLICY_REPO_CLONE_FOLDER_PREFIX,
+                                use_fixed_path=opal_server_config.POLICY_REPO_REUSE_CLONE_PATH
+                            )
+                            # only the leader should create new clone path and discard previous ones
+                            full_local_repo_path = clone_path_finder.create_new_clone_path()
+                            logger.info(f"Policy repo will be cloned to: {full_local_repo_path}")
+                            self.watcher = setup_watcher_task(self.publisher, remote_source_url=self._policy_remote_url, clone_path_finder=clone_path_finder)
```

Comment:
@roekatz the watcher lives under the assumption that the clone path is empty and the RepoCloner will simply clone into that directory.

What if i am a second leader selected after a previous leader already cloned and crashed?

I think the cloner should check:
- if there is already a git repo in the directory
- if the remote url is the same as the one we are trying to clone
- maybe if the git status is ok? (some check that git is in healthy state)
otherwise it should delete the contents and reclone.

Reply:
Yeah exactly that was my bug - already fixed it.
If the directory contains valid git repo (`Repo(dir)` doesn't raises) - then I don't try to clone it again.
Also added `checkout` to the right branch just to make sure.

Validating the remote_url actually makes sense, I haven't done it yet - checking if I can add it easily 
(actually it's probably not that important, if you change the remote url, you're going to re-run the container anyway.
the fact that all this logic only happens when inside docker, means it's basically only gonna occur on Gunicorn induced leader re-election).

Reply:
So in summary:
If the repo already exists on wrong branch, if would checkout the correct branch.
If the remote is wrong - is would raise a `GitError` for now.